typings/app.d.ts

```diff
@@ -76,6 +76,12 @@ declare namespace firebase {
     uid : string ;
   }
 
+  interface Observer<V, E> {
+    next?: (value: V | null) => any;
```

Comment:
In a few places inside the SDK, these are typed with a void return value (https://github.com/firebase/firebase-js-sdk/blob/657055d3c81101d55e86baac0a49d7cfd0a25863/src/app/subscribe.ts#L16, https://github.com/firebase/firebase-js-sdk/blob/657055d3c81101d55e86baac0a49d7cfd0a25863/src/storage/implementation/observer.ts#L18). Let's make the return types here "void" instead of "any" for consistency.